src/software/field_tests/field_test_fixture.py

```diff
@@ -0,0 +1,629 @@
+            team_robots = (
+                self.friendly_robot_ids_field
+                if idx == 0
+                else self.enemy_robot_ids_field
+            )
+            for robot_id in robot_ids:
+                if robot_id not in team_robots:
+                    logger.warning(
+                        f"robot {id} from the {team_name} team present in test but not on the field"
+                    )
+                    ids_present = False
+
+        if not ids_present:
+            raise Exception("robotIds do not match")
+
+    def __setBallState(self, worldstate: WorldState):
+        """Commands a robot to move the ball to the given position on the field
+
+        Args:
+            worldstate (worldState Proto): proto that contains ball position
+
+        Raises:
+            Exception: On failing to move the ball to the correct position
+        """
+
+        logger.info("starting ball placement")
+
+        # ball placement
+        if worldstate.HasField("ball_state"):
+
+            ball_position = tbots.createPoint(worldstate.ball_state.global_position)
+
+            dribble_tactic = DribbleTactic(
+                dribble_destination=worldstate.ball_state.global_position,
+                allow_excessive_dribbling=True,
+            )
+            move_ball_tactics = AssignedTacticPlayControlParams()
+            move_ball_tactics.assigned_tactics[
+                self.friendly_robot_ids_field[0]
+            ].dribble.CopyFrom(dribble_tactic)
+
+            self.blue_full_system_proto_unix_io.send_proto(
+                AssignedTacticPlayControlParams, move_ball_tactics
+            )
+
+            # validate completion
+            ball_placement_timout_s = 10
+            ball_placement_validation_function = BallEventuallyStopsInRegion(
+                regions=[tbots.Circle(ball_position, 0.1)]
+            )
+
+            try:
+                self.__validateWorldState(
+                    ball_placement_validation_function, ball_placement_timout_s
+                )
+            except:
+                raise Exception(
+                    "ball placement by blue robot {} to position {} failed".format(
+                        self.friendly_robot_ids_field[0], ball_position
+                    )
+                )
+
+    def __setRobotState(self, worldstate: WorldState):
+        """sets friendly and enemy robots to the given position on the field
+
+        Args:
+            worldstate (worldState Proto): proto that contains robot positions
+
+        Raises:
+            Exception: On failing to set robot positions
+        """
+        logger.info("moving robots to start position")
+
+        BLUE_TEAM_INDEX = 0
+        YELLOW_TEAM_INDEX = 1
+
+        # creating movement tactics and associated validation functions
+        initial_position_tactics = [
+            AssignedTacticPlayControlParams(),
+            AssignedTacticPlayControlParams(),
+        ]
```

Comment:
With 1 fullsystem, I dont think this function will fully work (i.e. We can't command the enemy robots to move)

Reply:
Removed